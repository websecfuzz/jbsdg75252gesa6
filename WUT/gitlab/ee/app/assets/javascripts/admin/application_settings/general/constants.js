import { s__ } from '~/locale';

export const DROPZONE_DESCRIPTION_TEXT = s__(
  'Licenses|Drag your license file here or %{linkStart}select a file to upload%{linkEnd}.',
);
export const FILE_UPLOAD_ERROR_MESSAGE = s__('Licenses|The file could not be uploaded.');
export const FILE_DROP_ERROR_MESSAGE = s__(
  'Licenses|Error: You are trying to upload something other than a file',
);
export const DROP_TO_START_MESSAGE = s__('Licenses|Drop your license file to start the upload.');
