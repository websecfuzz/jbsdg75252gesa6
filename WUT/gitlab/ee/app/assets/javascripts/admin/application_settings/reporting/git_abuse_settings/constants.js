import { s__, __, sprintf } from '~/locale';

export const MIN_NUM_REPOS = 0;
export const MAX_NUM_REPOS = 10000;

export const MIN_TIME_PERIOD = 0;
export const MAX_TIME_PERIOD = 10 * 24 * 60 * 60; // 10 days in seconds;

export const MAX_ALLOWED_USERS = 100;

export const MIN_ALERTED_USERS = 1;
export const MAX_ALERTED_USERS = 100;

export const NUM_REPOS_BLANK_ERROR = s__(
  "GitAbuse|Number of repositories can't be blank. Set to 0 for no limit.",
);
export const NUM_REPOS_NAN_ERROR = s__('GitAbuse|Number of repositories must be a number.');
export const NUM_REPOS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Number of repositories should be between %{minNumRepos}-%{maxNumRepos}.`),
  {
    minNumRepos: MIN_NUM_REPOS,
    maxNumRepos: MAX_NUM_REPOS,
  },
);

export const TIME_PERIOD_BLANK_ERROR = s__(
  "GitAbuse|Reporting time period can't be blank. Set to 0 for no limit.",
);
export const TIME_PERIOD_NAN_ERROR = s__('GitAbuse|Reporting time period must be a number.');
export const TIME_PERIOD_LIMIT_ERROR = sprintf(
  s__(
    `GitAbuse|Reporting time period should be between %{minTimePeriod}-%{maxTimePeriod} seconds.`,
  ),
  {
    minTimePeriod: MIN_TIME_PERIOD,
    maxTimePeriod: MAX_TIME_PERIOD,
  },
);

export const ALLOWED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|You cannot specify more than %{maxAllowedUsers} excluded users.`),
  {
    maxAllowedUsers: MAX_ALLOWED_USERS,
  },
);

export const NUM_REPO_LABEL = s__('GitAbuse|Number of repositories');
export const NUM_REPO_DESCRIPTION = s__(
  "GitAbuse|The maximum number of unique repositories a user can download in the specified time period before they're banned.",
);
export const REPORTING_TIME_PERIOD_LABEL = s__('GitAbuse|Reporting time period (seconds)');

export const ALLOWED_USERS_LABEL = s__('GitAbuse|Excluded users');
export const ALLOWED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are excluded from the Git abuse rate limit.',
);

export const ALERTED_USERS_LABEL = s__('GitAbuse|Send notifications to');
export const ALERTED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are emailed when Git abuse rate limit is exceeded.',
);
export const ALERTED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Select between %{minAlertedUsers} and %{maxAlertedUsers} users to notify.`),
  {
    minAlertedUsers: MIN_ALERTED_USERS,
    maxAlertedUsers: MAX_ALERTED_USERS,
  },
);

export const AUTO_BAN_TOGGLE_LABEL = s__(
  'GitAbuse|Automatically ban users from this %{scope} when they exceed the specified limits',
);

export const SEARCH_USERS = __('Search users');
export const SEARCH_TERM_TOO_SHORT = __('Enter at least three characters to search');
export const NO_RESULTS = __('No results');

export const SAVE_CHANGES = __('Save changes');
export const SUCCESS_MESSAGE = __('Application settings saved successfully.');

export const LOAD_ERROR_MESSAGE = __(
  'An error occurred while saving your settings. Try saving them again.',
);
export const SAVE_ERROR_MESSAGE = __(
  'An error occurred while retrieving your settings. Reload the page to try again.',
);
