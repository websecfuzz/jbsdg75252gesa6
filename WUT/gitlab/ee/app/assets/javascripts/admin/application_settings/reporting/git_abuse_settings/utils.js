import { parseBoolean } from '~/lib/utils/common_utils';

export const parseFormProps = ({
  maxNumberOfRepositoryDownloads,
  maxNumberOfRepositoryDownloadsWithinTimePeriod,
  gitRateLimitUsersAllowlist,
  gitRateLimitUsersAlertlist,
  autoBanUserOnExcessiveProjectsDownload,
}) => ({
  maxNumberOfRepositoryDownloads: parseInt(maxNumberOfRepositoryDownloads, 10),
  maxNumberOfRepositoryDownloadsWithinTimePeriod: parseInt(
    maxNumberOfRepositoryDownloadsWithinTimePeriod,
    10,
  ),
  gitRateLimitUsersAllowlist: JSON.parse(gitRateLimitUsersAllowlist),
  gitRateLimitUsersAlertlist: JSON.parse(gitRateLimitUsersAlertlist),
  autoBanUserOnExcessiveProjectsDownload: parseBoolean(autoBanUserOnExcessiveProjectsDownload),
});
