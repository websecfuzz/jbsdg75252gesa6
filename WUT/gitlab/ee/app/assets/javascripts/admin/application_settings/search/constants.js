import { __, s__ } from '~/locale';

export const MINIMUM_QUERY_LENGTH = 3;
export const NO_RESULTS_TEXT = __('No results found.');
export const SEARCH_QUERY_TOO_SHORT = __('Enter at least three characters to search.');
export const ENTITIES_FETCH_ERROR = s__(
  'AdvancedSearch|Could not fetch index entities. Please try again later.',
);
