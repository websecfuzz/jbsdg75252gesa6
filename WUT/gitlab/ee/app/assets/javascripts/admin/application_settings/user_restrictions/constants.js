import { helpPagePath } from '~/helpers/help_page_helper';

export const PRIVATE_PROFILES_DISABLED_ICON = 'private-profiles-disabled-icon';
export const PRIVATE_PROFILES_DISABLED_HELP_LINK = helpPagePath(
  'administration/settings/account_and_limit_settings',
  { anchor: 'set-profiles-of-new-users-to-private-by-default' },
);
