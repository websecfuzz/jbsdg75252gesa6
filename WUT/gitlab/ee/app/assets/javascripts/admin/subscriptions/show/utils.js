import {
  subscriptionTypes,
  offlineCloudLicenseText,
  onlineCloudLicenseText,
  licenseFileText,
} from './constants';

export const getLicenseFromData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.license;
export const getErrorsAsData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.errors || [];

export function getLicenseTypeLabel(type) {
  switch (type) {
    case subscriptionTypes.OFFLINE_CLOUD:
      return offlineCloudLicenseText;
    case subscriptionTypes.ONLINE_CLOUD:
      return onlineCloudLicenseText;
    default:
      return licenseFileText;
  }
}
