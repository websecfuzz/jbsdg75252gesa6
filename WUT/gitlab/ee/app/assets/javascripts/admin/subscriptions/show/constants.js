import { __, s__ } from '~/locale';
import { PROMO_URL } from '~/constants';

export const subscriptionMainTitle = s__('SuperSonics|Instance subscription');
export const subscriptionActivationNotificationText = s__(
  `SuperSonics|Your subscription was successfully activated. You can see the details below.`,
);
export const subscriptionActivationFutureDatedNotificationTitle = s__(
  'SuperSonics|Your future dated license was successfully added',
);
export const subscriptionActivationFutureDatedNotificationMessage = s__(
  'SuperSonics|You have successfully added a license that activates on %{date}. Please see the subscription history table below for more details.',
);
export const subscriptionActivationInsertCode = __(
  'Find the activation code for a purchased or renewed subscription in the Customers Portal or purchase confirmation email.',
);
export const howToActivateSubscription = s__(
  'SuperSonics|Learn how to %{linkStart}activate your subscription%{linkEnd}.',
);
export const subscriptionHistoryFailedTitle = s__('SuperSonics|Subscription unavailable');
export const subscriptionHistoryFailedMessage = s__(
  'SuperSonics|Your %{subscriptionEntryName} cannot be displayed at the moment. Please refresh the page to try again.',
);
export const currentSubscriptionsEntryName = s__('SuperSonics|current subscription');
export const pastSubscriptionsEntryName = s__('SuperSonics|past subscriptions');
export const futureSubscriptionsEntryName = s__('SuperSonics|future subscriptions');

export const cancelLabel = __('Cancel');
export const activateLabel = s__('AdminUsers|Activate');
export const activateSubscription = s__('SuperSonics|Activate subscription');
export const addActivationCode = s__('SuperSonics|Add activation code');
export const noActiveSubscription = s__(`SuperSonics|You do not have an active subscription`);
export const subscriptionDetailsHeaderText = s__('SuperSonics|Subscription details');
export const licensedToHeaderText = s__('SuperSonics|Licensed to');
export const copySubscriptionIdButtonText = __('Copy');
export const licenseFileText = __('Legacy license');
export const onlineCloudLicenseText = s__('SuperSonics|Online license');
export const offlineCloudLicenseText = s__('SuperSonics|Offline license');
export const detailsLabels = {
  address: __('Address'),
  company: __('Company'),
  email: __('Email'),
  id: __('ID'),
  lastSync: s__('Subscriptions|Last sync'),
  name: licensedToHeaderText,
  plan: __('Plan'),
  type: __('Type'),
  expiresAt: s__('Subscriptions|End date'),
  startsAt: s__('Subscriptions|Start date'),
};

export const subscriptionTable = {
  activatedAt: s__('Subscriptions|Activation date'),
  expiresOn: s__('Subscriptions|End date'),
  seats: __('Seats'),
  startsAt: s__('Subscriptions|Start date'),
  title: __('Subscription History'),
  type: __('Type'),
};

export const subscriptionActivationForm = {
  activationCode: s__('SuperSonics|Activation code'),
  activationCodeFeedback: s__(
    'SuperSonics|The activation code should be a 24-character alphanumeric string',
  ),
  pasteActivationCode: s__('SuperSonics|Enter activation code'),
  acceptTerms: s__(
    'SuperSonics|I agree that my use of the GitLab Software is subject to the Subscription Agreement located at the %{linkStart}Terms of Service%{linkEnd}, unless otherwise agreed to in writing with GitLab.',
  ),
  acceptTermsFeedback: s__('SuperSonics|Select that you agree before activating the subscription.'),
};

export const subscriptionSyncStatus = {
  SYNC_FAILURE: 'SYNC_FAILURE',
  SYNC_PENDING: 'SYNC_PENDING',
  SYNC_SUCCESS: 'SYNC_SUCCESS',
};

export const subscriptionTypes = {
  ONLINE_CLOUD: 'online_cloud',
  OFFLINE_CLOUD: 'offline_cloud',
  LEGACY_LICENSE: 'legacy_license',
};

export const trialCard = {
  title: s__('SuperSonics|Free trial'),
  description: s__(
    'SuperSonics|You can start a free trial of GitLab Ultimate without any obligation or payment details.',
  ),
  startTrial: s__('SuperSonics|Start free trial'),
};

export const buySubscriptionCard = {
  title: __('Subscription'),
  description: s__(
    'SuperSonics|Ready to get started? A GitLab plan is ideal for scaling organizations and for multi team usage.',
  ),
  buttonLabel: s__('SuperSonics|Buy subscription'),
};

export const SUBSCRIPTION_ACTIVATION_FAILURE_EVENT = 'subscription-activation-failure';
export const SUBSCRIPTION_ACTIVATION_SUCCESS_EVENT = 'subscription-activation-success';
export const SUBSCRIPTION_ACTIVATION_FINALIZED_EVENT = 'subscription-activation-finalized';

// Server-sent error messages (or regexes to match those messages) to detect error type
export const INVALID_ACTIVATION_CODE_SERVER_ERROR = 'invalid activation code';
export const EXPIRED_LICENSE_SERVER_ERROR = 'This license has already expired.';
export const SILENT_MODE_ENABLED_ERROR =
  'only get, head, options, and trace methods are allowed in silent mode';
export const SUBSCRIPTION_NOT_FOUND_SERVER_ERROR = 'without cloud compatible subscription';
export const SUBSCRIPTION_OVERAGES_SERVER_ERROR_REGEX =
  /(This GitLab installation currently has|During the year before this license started, this GitLab installation had) (\d+) active users?, exceeding this license's limit of (\d+) by (\d+) users?\. Please add a license for at least (\d+) users? or contact sales at https:\/\/about\.gitlab\.com\/sales\//;
export const SUBSCRIPTION_INSUFFICIENT_TRUE_UP_SERVER_ERROR_REGEX =
  /You have applied a True-up for (\d+) users? but you need one for (\d+) users?\. Please contact sales at https:\/\/about\.gitlab\.com\/sales\//;

export const CONNECTIVITY_ERROR = 'CONNECTIVITY_ERROR';
export const INVALID_CODE_ERROR = 'INVALID_CODE_ERROR';

export const supportLink = `${PROMO_URL}/support/#contact-support`;
export const exportLicenseUsageBtnText = s__('SuperSonics|Export license usage file');
export const customersPortalBtnText = s__('SuperSonics|Customers Portal');

export const instanceHasFutureLicenseBanner = {
  title: s__('SuperSonics|You have a future dated license'),
  message: s__(
    'SuperSonics|You have added a license that activates on %{date}. Please see the subscription history table below for more details.',
  ),
};

// Subscription Sync Button
export const SYNC_BUTTON_ID = 'syncButton';
export const syncButtonTexts = Object.freeze({
  syncSubscriptionButtonText: s__('SuperSonics|Sync subscription details'),
  syncSubscriptionTooltipText: s__(
    'SuperSonics|You can sync your subscription data to ensure your details are up to date.',
  ),
});

export const VIEW_ADMIN_SUBSCRIPTION_PAGELOAD = 'view_admin_subscription_pageload';
