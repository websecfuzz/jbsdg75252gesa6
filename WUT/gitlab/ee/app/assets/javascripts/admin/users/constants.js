import { s__ } from '~/locale';
import { getTokenConfigs } from '~/admin/users/constants';

export {
  SOLO_OWNED_ORGANIZATIONS_EMPTY,
  I18N_USER_ACTIONS,
  SOLO_OWNED_ORGANIZATIONS_REQUESTED_COUNT,
} from '~/admin/users/constants';

export const TOKEN_CONFIGS = getTokenConfigs([
  { value: 'admins', title: s__('AdminUsers|Administrator') },
  { value: 'auditors', title: s__('AdminUsers|Auditor') },
  { value: 'external', title: s__('AdminUsers|External') },
]);
