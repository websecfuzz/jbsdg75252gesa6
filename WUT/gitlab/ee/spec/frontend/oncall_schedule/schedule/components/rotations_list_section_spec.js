import { GlCard } from '@gitlab/ui';
import ScheduleShiftWrapper from 'ee/oncall_schedules/components/schedule/components/shifts/components/schedule_shift_wrapper.vue';
import CurrentDayIndicator from 'ee/oncall_schedules/components/schedule/components/current_day_indicator.vue';
import RotationsListSection from 'ee/oncall_schedules/components/schedule/components/rotations_list_section.vue';
import { getTimeframeForWeeksView } from 'ee/oncall_schedules/components/schedule/utils';
import { PRESET_TYPES } from 'ee/oncall_schedules/constants';
import { useFakeDate } from 'helpers/fake_date';
import { createMockDirective, getBinding } from 'helpers/vue_mock_directive';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { scheduleIid } from '../../mocks/apollo_mock';
import mockRotations from '../../mocks/mock_rotation.json';

jest.mock('~/lib/utils/color_utils');

describe('RotationsListSectionComponent', () => {
  let wrapper;
  const mockTimeframeInitialDate = new Date(mockRotations[0].shifts.nodes[0].startsAt);
  const mockTimeframeWeeks = getTimeframeForWeeksView(mockTimeframeInitialDate);

  function createComponent({
    presetType = PRESET_TYPES.WEEKS,
    timeframe = mockTimeframeWeeks,
    userCanCreateSchedule = true,
  } = {}) {
    wrapper = shallowMountExtended(RotationsListSection, {
      propsData: {
        presetType,
        timeframe,
        scheduleIid,
        rotations: [mockRotations[0]],
      },
      provide: {
        userCanCreateSchedule,
      },
      stubs: {
        GlCard,
      },
      directives: {
        GlTooltip: createMockDirective('gl-tooltip'),
      },
    });
  }

  const findTimelineCells = () => wrapper.findAllByTestId('timeline-cell');
  const findCurrentDayIndicatorContent = () => wrapper.findByTestId('current-day-indicator');
  const findCurrentDayIndicatorRotations = () =>
    wrapper.findByTestId('current-day-indicator-with-rotations');
  const findAllScheduleShiftWrappers = () => wrapper.findAllComponents(ScheduleShiftWrapper);
  const findRotationName = (id) => wrapper.findByTestId(`rotation-name-${id}`);
  const findRotationNameTooltip = (id) => getBinding(findRotationName(id).element, 'gl-tooltip');
  const findEditAndDeleteButtons = () => wrapper.findByTestId('rotation-edit-button-group');

  describe('when the timeframe includes today', () => {
    useFakeDate(2021, 0, 14);

    beforeEach(() => {
      createComponent();
    });

    it('renders component layout', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the current day indicator if the timeframe includes the current day', () => {
      expect(findCurrentDayIndicatorRotations().exists()).toBe(true);
    });

    it('renders timeline cell items based on timeframe data', () => {
      const mockTimelineCellWidth = 1;
      expect(findTimelineCells()).toHaveLength(mockTimelineCellWidth);
    });

    it('renders current day indicator in the first timeline cell', () => {
      expect(findTimelineCells().at(0).findComponent(CurrentDayIndicator).exists()).toBe(true);
    });

    it('render the correct amount of rotation assignees with their related information', () => {
      expect(findAllScheduleShiftWrappers()).toHaveLength(1);
      expect(findAllScheduleShiftWrappers().at(0).props('rotation')).toEqual(mockRotations[0]);
    });

    it('renders a tooltip with the rotation name', () => {
      const rotationNameTT = findRotationNameTooltip(mockRotations[0].id);
      expect(rotationNameTT.value).toBeDefined();
      expect(rotationNameTT.value.title).toBe(mockRotations[0].name);
    });

    it('renders edit and delete buttons', () => {
      expect(findEditAndDeleteButtons().exists()).toBe(true);
    });
  });

  describe('when the timeframe does not include today', () => {
    useFakeDate(2021, 0, 31);

    beforeEach(() => {
      createComponent();
    });

    it('does not render the current day indicator', () => {
      expect(findCurrentDayIndicatorContent().exists()).toBe(false);
    });
  });

  describe('when user cannot create schedule', () => {
    beforeEach(() => {
      createComponent({ userCanCreateSchedule: false });
    });

    it('does not render edit and delete buttons', () => {
      expect(findEditAndDeleteButtons().exists()).toBe(false);
    });
  });
});
