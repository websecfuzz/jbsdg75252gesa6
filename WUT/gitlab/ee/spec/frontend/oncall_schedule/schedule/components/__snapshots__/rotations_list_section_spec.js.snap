// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotationsListSectionComponent when the timeframe includes today renders component layout 1`] = `
<div
  class="list-section"
>
  <div>
    <div
      class="gl-border-t gl-clearfix"
    >
      <span
        class="details-cell gl-flex gl-float-left gl-items-center gl-justify-between gl-left-0 gl-sticky"
      >
        <span
          aria-label="Rotation 242"
          class="gl-text-subtle gl-truncate"
          data-testid="rotation-name-gid://gitlab/IncidentManagement::OncallRotation/2"
        >
          Rotation 242
        </span>
        <div
          class="gl-px-2"
          data-testid="rotation-edit-button-group"
        >
          <gl-button-stub
            aria-label="Edit rotation"
            buttontextclasses=""
            category="tertiary"
            icon="pencil"
            role="button"
            size="medium"
            tabindex="0"
            tag="button"
            title="Edit rotation"
            type="button"
            variant="default"
          />
          <gl-button-stub
            aria-label="Delete rotation"
            buttontextclasses=""
            category="tertiary"
            icon="remove"
            role="button"
            size="medium"
            tabindex="0"
            tag="button"
            title="Delete rotation"
            type="button"
            variant="default"
          />
        </div>
      </span>
      <span
        class="gl-float-left gl-overflow-hidden gl-relative timeline-cell"
        data-testid="timeline-cell"
        style="width: calc(100% - 180px);"
      >
        <current-day-indicator-stub
          data-testid="current-day-indicator-with-rotations"
          presettype="WEEKS"
          timeframeitem="Tue Jan 12 2021 00:00:00 GMT+0000 (Greenwich Mean Time)"
          timelinewidth="2"
        />
        <schedule-shift-wrapper-stub
          presettype="WEEKS"
          rotation="[object Object]"
          timeframe="Tue Jan 12 2021 00:00:00 GMT+0000 (Greenwich Mean Time),Tue Jan 19 2021 00:00:00 GMT+0000 (Greenwich Mean Time)"
        />
      </span>
    </div>
  </div>
</div>
`;
